src/portfolio/APIClient.ts

```diff
@@ -72,6 +72,19 @@ export type KeyFigures = {
   dividendPayoutRatio: number;
 };
 
+export type Correlations = {
+  [key: string]: number;
+};
```

Comment:
Would it make sense to convert to a more useful data format here?

Reply:
That is being done with the MappedCorrelation type.

Reply:
Could you please briefly explain how that data conversion works? I'm curious why converting data from a structure to another is so complex.

If it can't be greatly simplified, I think it should at least me moved outside the rendering logic of the heatmap component in a dedicated file.